prow/plugins/approve/approvers/owners.go

```diff
@@ -599,10 +612,10 @@ The pull request process is described [here](https://git.k8s.io/community/contri
 <details {{if (and (not .ap.AreFilesApproved) (not (call .ap.ManuallyApproved))) }}open{{end}}>
 Needs approval from an approver in each of these files:
 
-{{range .ap.GetFiles .org .project}}{{.}}{{end}}
+{{range .ap.GetFiles .org .project .branch}}{{.}}{{end}}
```

Comment:
I just realized I never wanted to see how complex this template was :scream:

Reply:
The approval plugin needs to be redone. The logic is overly complex, inefficient, and doesn't work with regexp filtered OWNERS files. Before reworking the plugin though, I need to determine how approver recommendations will work with regexp filtered OWNERS files now that OWNERS files are not directory scoped.
Might be worth a break out discussion at some point.

Reply:
honestly still not as bad as the templates in prow config :+)
obviously test-infra just needs it's own template language /s

Reply:
also, fixing approval for regexp would be *great*, finer grained approval would be really helpful in k/k